test/functional/wallet-hd.py

```diff
@@ -91,6 +91,21 @@ def run_test (self):
         self.start_node(1, extra_args=self.extra_args[1] + ['-rescan'])
         assert_equal(self.nodes[1].getbalance(), num_hd_adds + 1)
 
+        # Try a RPC based rescan
+        self.stop_node(1)
+        shutil.rmtree(tmpdir + "/node1/regtest/blocks")
+        shutil.rmtree(tmpdir + "/node1/regtest/chainstate")
+        shutil.copyfile(tmpdir + "/hd.bak", tmpdir + "/node1/regtest/wallet.dat")
+        self.start_node(1, extra_args=self.extra_args[1])
+        connect_nodes_bi(self.nodes, 0, 1)
+        self.sync_all()
+        out = self.nodes[1].rescanblockchain(0,1)
+        assert_equal(out['startheight'], 0)
+        assert_equal(out['stopheight'], 1)
+        out = self.nodes[1].rescanblockchain()
+        assert_equal(out['startheight'], 0)
+        assert_equal(self.nodes[1].getbalance(), num_hd_adds + 1)
```

Comment:
Perhaps assert that the rescan scanned up to the tip:

`assert_equal(out['stopheight'], self.nodes[1].getblockcount())`